### ISSUE TITLE
Can not reproduce the results of the paper

### ISSUE THREAD

**Issue by sndnyang** [yanlirock/RLANS#7]
Hi, 

I use your code to retrain the model, but I do not get the result in the paper. I just follow your instruction.
1) split 
2) training the language model
3) python Adversarial_training.py --cuda --lr=0.001 --batch_size=128 --save='./ag_adv_model/' --pre_train='./ag_lm_model' --number_per_class=1000 --reduce_rate=0.95

But I can only get about 90%
```
Accuracy of the classifier on the test data is: 89.9855
```
Can you tell me how to set parameters?

I use PyTorch 1.3 or PyTorch 1.1. I think the version won't make this gap.
Thank you!

**Comment by yanlirock** [comment]
You can compare with the 
"python classifier_training.py --cuda --lr=0.001 --batch_size=128 --save='/classify_no_pre/' --pre_train='' --number_per_class=1000 --reduce_rate=0.95" to check the difference between Adversarial_training.py and classifier_training.py under the same condition. The results showed in the paper is a tensorflow version, which is very nasty with a lot of hard code. And hence I don't upload it.

**Comment by yanlirock** [comment]
You can compare with the
python classifier_training.py --cuda --lr=0.001 --batch_size=128 --save='/classify_with_pre/' --pre_train='/ag_lm_model' --number_per_class=1000 --reduce_rate=0.95

**Comment by sndnyang** [comment]
> You can compare with the
> python classifier_training.py --cuda --lr=0.001 --batch_size=128 --save='/classify_with_pre/' --pre_train='/ag_lm_model' --number_per_class=1000 --reduce_rate=0.95

Thank you! One more question,  this result of  classify_with_pre represents the  "SeqSSL" result in the paper,   It's not SeqSSL+AT/VAT/SELF. right?

**Comment by yanlirock** [comment]
yes

**Comment by huanglx27** [comment]
> You can compare with the
> "python classifier_training.py --cuda --lr=0.001 --batch_size=128 --save='/classify_no_pre/' --pre_train='' --number_per_class=1000 --reduce_rate=0.95" to check the difference between Adversarial_training.py and classifier_training.py under the same condition. The results showed in the paper is a tensorflow version, which is very nasty with a lot of hard code. And hence I don't upload it.

 "python classifier_training.py --cuda --lr=0.001 --batch_size=128 --save='/classify_no_pre/' --pre_train='' --number_per_class=1000 --reduce_rate=0.95" 
I tried this but can only get about 30%, while others seems to be normal.

**Comment by yanlirock** [comment]
> > You can compare with the
> > "python classifier_training.py --cuda --lr=0.001 --batch_size=128 --save='/classify_no_pre/' --pre_train='' --number_per_class=1000 --reduce_rate=0.95" to check the difference between Adversarial_training.py and classifier_training.py under the same condition. The results showed in the paper is a tensorflow version, which is very nasty with a lot of hard code. And hence I don't upload it.
> 
> "python classifier_training.py --cuda --lr=0.001 --batch_size=128 --save='/classify_no_pre/' --pre_train='' --number_per_class=1000 --reduce_rate=0.95"
> I tried this but can only get about 30%, while others seems to be normal.

this is no pretrained model. so the accuracy should be lowest.

**Comment by sndnyang** [comment]
> > "python classifier_training.py --cuda --lr=0.001 --batch_size=128 --save='/classify_no_pre/' --pre_train='' --number_per_class=1000 --reduce_rate=0.95"
> > I tried this but can only get about 30%, while others seems to be normal.
> 
> this is no pretrained model. so the accuracy should be lowest.

OK, I got it.  But you didn't show a result based on this no pre-trained model in Table 3 in your paper. 
And your three models in the repo give different results.

The no pre-trained model with 1000 per class gets <30% on AG news, while your LSTM result in Figure 5 is about 40%.
The pre-trained model can get 88.7% at epoch 100, while the corresponding model(SeqSSL) in Table 3 is about 87.3%
The RLANS model gets ~ 90%, while your result(RLANS) in Table 3 in the paper is 92.5%.

I know you use TF to get the results in the paper, and the repo uses PyTorch.  I suggest you update the readme about the results.  I'm really confused about the results of whether I reproduce it successfully.

BTW, a type: 4)For comparison, you can train the semisupervised sequence learning model (SSL), without the pretrained language model       # 'without'  should be 'with'
And writing the reference in the bibtex is better, right? From google scholar:
```
@inproceedings{ssltcpolicy2018,
  title={Learning adversarial networks for semi-supervised text classification via policy gradient},
  author={Li, Yan and Ye, Jieping},
  booktitle={Proceedings of the 24th ACM SIGKDD International Conference on Knowledge Discovery \& Data Mining (KDD 2018)},
  year={2018},
}
```
Thank you for your reply.

**Comment by huanglixiao** [comment]
> > > "python classifier_training.py --cuda --lr=0.001 --batch_size=128 --save='/classify_no_pre/' --pre_train='' --number_per_class=1000 --reduce_rate=0.95"
> > > I tried this but can only get about 30%, while others seems to be normal.
> > 
> > 
> > this is no pretrained model. so the accuracy should be lowest.
>  I suggest you update the readme about the results. 

great suggestion.

**Comment by huanglx27** [comment]
> > > "python classifier_training.py --cuda --lr=0.001 --batch_size=128 --save='/classify_no_pre/' --pre_train='' --number_per_class=1000 --reduce_rate=0.95"
> > > I tried this but can only get about 30%, while others seems to be normal.
> > 
> > 
> > this is no pretrained model. so the accuracy should be lowest.
> 
>  I suggest you update the readme about the results. I'm really confused about the results of whether I reproduce it successfully.

great suggestion

Actually, LSTM result in Table3 is 75.58%...
